src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepExecution.java

```diff
@@ -56,6 +56,7 @@ public boolean start() throws Exception {
         newTemplate.setContainers(step.getContainers());
         newTemplate.setNodeSelector(step.getNodeSelector());
         newTemplate.setServiceAccount(step.getServiceAccount());
+        newTemplate.setAnnotations(step.getAnnotations());
```

Comment:
FYI when upgrading getAnnotations() is going to be null, because is loaded from xml config where it doesn't exist, have you checked that the API won't break with a null value?